### PR TITLE
remove audio permission, upgraded dependencies, and fix ide warnings

### PR DESCRIPTION
**Pull request by zhaonian**
- remove audio permission, not needed for CameraXBasic?
- upgrade dependencies
- increased `targetSdkcVersion` to latest 29

### CODE REVIEW

File: CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/PermissionsFragment.kt [hunk 1]
```diff
@@ -55,7 +52,7 @@ class PermissionsFragment : Fragment() {
             requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults)
         if (requestCode == PERMISSIONS_REQUEST_CODE) {
-            if (grantResults.size > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
```
Comment (owahltinez) [inline CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/PermissionsFragment.kt]: How about grantResults.firstOrNull()?

File: CameraXBasic/build.gradle [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import kotlin.KotlinVersion
```
Comment (owahltinez) [inline CameraXBasic/build.gradle]: Move after the copyright block

File: CameraXBasic/build.gradle [hunk 3]
```diff
@@ -18,17 +20,17 @@
 
 buildscript {
     // Top-level variables used for versioning
-    ext.kotlin_version = '+'
+    ext.kotlin_version = KotlinVersion.CURRENT
```
Comment (owahltinez) [inline CameraXBasic/build.gradle]: How is this better / different than '+'?

File: CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/PermissionsFragment.kt (hunk 1 above)
Comment (zhaonian) [reply CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/PermissionsFragment.kt]: Great!

File: CameraXBasic/build.gradle (hunk 3 above)
Comment (zhaonian) [reply CameraXBasic/build.gradle]: none I can think of :/ 
reverted this change

**Pull request merged**